test/WebJobs.Script.Tests/Controllers/ControllerScenarioTestFixture.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web.Http;
+using Autofac;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Controllers
+{
+    public class ControllerScenarioTestFixture
+    {
+        private HttpConfiguration _config;
+
+        public ControllerScenarioTestFixture()
+        {
+            _config = new HttpConfiguration();
+
+            HostSettings = new WebHostSettings
+            {
+                IsSelfHost = true,
+                ScriptPath = Path.Combine(Environment.CurrentDirectory, @"..\..\..\..\sample"),
+                LogPath = Path.Combine(Path.GetTempPath(), @"Functions"),
+                SecretsPath = Path.Combine(Environment.CurrentDirectory, @"..\..\..\..\src\WebJobs.Script.WebHost\App_Data\Secrets")
+            };
+
+            WebApiConfig.Register(_config, HostSettings, RegisterDependencies);
+
+            HttpServer = new HttpServer(_config);
+            this.HttpClient = new HttpClient(HttpServer);
+            this.HttpClient.BaseAddress = new Uri("https://localhost/");
+
+            WaitForHost();
```

Comment:
I checked this when I put the tests in place and the cost didn't seem significant. We could use a shared collection class fixture that is responsible for the host and shared across many test classes as well.